### PR TITLE
NomadTasker unit tests 

### PR DESCRIPTION
**Pull request by meganokeefe**
Added unit tests for the NomadTasker. Uses a mock database and a mock NomadWrapper.

- [x] wrapped all `nomad.go` API Client functions into `nomad_wrapper` functions 
- [x] mocked those interface functions with mockgen
- [x] implemented those interface functions for the real Nomad API in `nomad_wrapper/nomad_api.go` 
- [x] `nomad_test.go` tests all NomadTasker functions with no dependencies

### CODE REVIEW

File: task/nomad.go [hunk 1]
```diff
@@ -32,29 +33,25 @@ func NewNomadConfig() NomadConfig {
...
-	conf := nomadapi.DefaultConfig()
-	conf.Address = inConf.ClientUrl
-	if !strings.Contains(conf.Address, "http") {
-		conf.Address = fmt.Sprintf("http://%s", conf.Address)
-	}
-	nc, err := nomadapi.NewClient(conf)
+	nc, err := wrapper.NewNomadClient(inConf.ClientUrl)
 	if err != nil {
-		return &NomadTasker{}, err
+		return &NomadTasker{}, nil
```
Comment (stephenrlouie) [inline task/nomad.go]: why return nil here if it fails to make the client?

File: task/nomad_test.go [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+	err := nt.Remove(testOp)
+	if err != nil {
+		t.Errorf("%s Remove failure: %v", failed, err)
+	} else {
+		t.Logf("%s Successful Remove", succeed)
+	}
+}
+
+func TestNomadStatus(t *testing.T) {
+	ctrl := gomock.NewController(t)
```
Comment (stephenrlouie) [inline task/nomad_test.go]: Is it worth testing all status returns?

File: task/nomad_test.go [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+	} else {
+		if stat != model.ServiceOperationStatusRunning {
+			t.Errorf("%s Got unexpected status: %s", failed, stat)
+		}
+		t.Logf("%s Successful get Status --> %s", succeed, stat)
+	}
+}
+
+func TestNomadOutputs(t *testing.T) {
+	ctrl := gomock.NewController(t)
```
Comment (stephenrlouie) [inline task/nomad_test.go]: could test a DB failure and a successful Getshipper

File: task/nomad_test.go [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+		if !reflect.DeepEqual(expectOutput, outputResult) {
+			t.Errorf("%s Expected outputs: \n%v, but got:\n%v", failed, expectOutput, outputResult)
+		} else {
+			t.Logf("%s Successful get Output--> %v", succeed, outputResult)
+		}
+	}
+}
+
+func TestNomadProgress(t *testing.T) {
+	ctrl := gomock.NewController(t)
```
Comment (stephenrlouie) [inline task/nomad_test.go]: failure case and successful one

File: task/nomad_test.go [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+		if progResult != "deploy" {
+			t.Errorf("%s Unexpected Progress result: %s", failed, progResult)
+		} else {
+			t.Logf("%s Successful get Progress --> %s", succeed, progResult)
+		}
+	}
+}
+
+func TestNomadLogs(t *testing.T) {
+	ctrl := gomock.NewController(t)
```
Comment (stephenrlouie) [inline task/nomad_test.go]: fail & pass case

File: task/nomad_test.go (hunk 3 above)
Comment (stephenrlouie) [inline task/nomad_test.go]: I know this is changing very soon, so possibly not worth it.

Comment (meganokeefe) [inline task/nomad_test.go]: i'm literally deleting this function from tasker right now, so i'm gonna hold off on adding to the logs/progress/output tests.

**Pull request merged**